### PR TITLE
Port some audio code from SDL2 Mixer to SDL2

### PR DESCRIPTION
**Pull request by Ghabry**
The frequency change code was missing.

Also changed the emscripten format to F32, this is the native format and maybe a bit faster.

### CODE REVIEW

**Comment by Ghabry** [comment]
okay this needs more testing, sounds a bit broken xD

File: src/audio_sdl.cpp [hunk 1]
```diff
@@ -70,9 +74,27 @@ SdlAudio::SdlAudio() :
...
+#elif defined(EMSCRIPTEN)
+	// Get preferred sample rate from Browser (-> OS)
+	const int frequency = EM_ASM_INT_V({
+		var context;
+		try {
+			context = new AudioContext();
+		} catch (e) {
+			context = new webkitAudioContext();
+		}
+		return context.sampleRate;
```
Comment (fmatthew5876) [inline src/audio_sdl.cpp:89]: Isn't this a memory leak?

Comment (Ghabry) [reply src/audio_sdl.cpp:89]: This is JavaScript code so will be garbage collected.

**Comment by carstene1ns** [comment]
I guess this needs a test on Wii. In audio playback this should not be really noticeable, however general responsiveness should be better.

**Comment by Ghabry** [comment]
The F32 audio is obviously broken when using SDL (#2619) so I dropped this for now and only added the samplerate part

**Review by fdelapena — APPROVED** [review]

**Review by carstene1ns — APPROVED** [review]

**Pull request merged**